### DISCUSSION TITLE
Board support packages for custom STM32L010K4 microcontroller board

### DISCUSSION THREAD

**Discussion by mishraEmb** [stm32duino/Arduino_Core_STM32#1753] [Q&A] [answered]
In compatible with Arduino IDE, there are few stm32 series based micro board packages are available. Here I am looking for stm32L010K4T6 micro based custom board package which is available in category of variants support. If I can access in Arduino IDE. Please provide any suggestions.

**Comment by fpistm** [comment, marked as answer]
Hi @mishraEmb 

First, please avoid to open an issue for such kind of question.
We provide list of files for each mcu in variant (automatically generated) but it requires to declare it manually in the boards.txt and to add a linker scripts. Some of them are already done but not all. This is the case for the STM32L010K4.
You can refers to the wiki to add it:
https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-%28board%29#define-a-new-generic-variant

**Reply by mishraEmb** [reply]
Hello @fpistm 
After adding ldsccript.ld file into variant folder and making changes of (Adding custom board info) board.txt file, I am able to list that custom board in Arduino IDE but after compilation I am getting this following error.
"**cannot open linker script file C:\Users\Admin\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.2.0\variants\STM32L0xx\L010K4T_L011K(3-4)T_L021K4T/ldscript.ld: No such file or directory**"
Do I have to add this linker script file path to any other location?

**Reply by fpistm** [reply]
Is the file properly named ? as you wrote : `ldsccript.ld` instead of `ldscript.ld`

**Reply by mishraEmb** [reply]
Sorry for typos here, but I have renamed correctly in variant folder as you can see in the attached image.
![image](https://user-images.githubusercontent.com/108390197/176636725-41bb82b6-86de-4852-88ee-cb792a426b23.png)

**Reply by fpistm** [reply]
Well in that case I don't know what happen. Maybe permissions issue on the file?

**Reply by mishraEmb** [reply]
@fpistm 
Thank you for your reply, I have solved that issue by changing the ldscript.ld file as a text extension to LD extension and it is compiling correctly but due to less RAM and Flash size of STM32L010K4 controller, showing memory overflow. For  memory optimization which process would be more suitable by using LL_API or using CMSIS micro register level programming.
Thank you.

**Reply by fpistm** [reply]
ah good catch, I didn't see you explorer doesn't show file extension.
About size you can try yo reduce the code by disabling unused feature:
https://github.com/stm32duino/wiki/wiki/HAL-configuration#list-of-hal__module_disabled-definition
https://github.com/stm32duino/wiki/wiki/Custom-definitions#custom-pinmap-array
